test/e2e/specs/specs-playwright/wp-signup__paid.ts

```diff
@@ -0,0 +1,183 @@
+			await domainSearchComponent.search( blogName );
+			await domainSearchComponent.selectDomain( targetDomain );
+		} );
+
+		it( 'Select WordPress.com Personal plan', async function () {
+			const signupPickPlanPage = new SignupPickPlanPage( page );
+			await signupPickPlanPage.selectPlan( 'Personal' );
+		} );
+
+		it( 'See secure payment', async function () {
+			cartCheckoutPage = new CartCheckoutPage( page );
+			await cartCheckoutPage.validateCartItem( 'WordPress.com Personal' );
+			// await cartCheckoutPage.validateCartItem( targetDomain );
+		} );
+
+		it( 'Prices are shown with expected currency symbol', async function () {
+			const buttonText = await cartCheckoutPage.getPaymentButtonText();
+			await buttonText.startsWith( '£' );
+		} );
+
+		it( 'Remove domain purchase from cart', async function () {
+			await cartCheckoutPage.removeCartItem( targetDomain );
+		} );
+
+		it( 'Apply coupon and validate resulting purchase amount', async function () {
+			const originalAmount = await cartCheckoutPage.getCheckoutTotalAmount();
+			await cartCheckoutPage.enterCouponCode();
+			const newAmount = await cartCheckoutPage.getCheckoutTotalAmount();
+
+			expect( newAmount ).toBeLessThan( originalAmount );
+			const expectedAmount = originalAmount * 0.99;
+
+			// Some currencies do not typically have decimal places.
+			// eg. USD would commonly have 2 decimal places, e.g. 12.34.
+			// In JPY or TWD there will be no decimal digits.
+			// Drop decimals so that the result won't be affected by the currency variation.
+			expect( Math.floor( newAmount ) ).toStrictEqual( Math.floor( expectedAmount ) );
+		} );
+
+		it( 'Enter billing and payment details', async function () {
+			const paymentDetails = DataHelper.getTestPaymentDetails( 'Visa' ) as PaymentDetails;
+			await cartCheckoutPage.enterBillingDetails( paymentDetails );
+			await page.pause();
+			await cartCheckoutPage.enterPaymentDetails( paymentDetails );
+		} );
+
+		it( 'Make purchase', async function () {
+			await cartCheckoutPage.purchase();
+		} );
+
+		it( 'Decline Quickstart session', async function () {
+			// No dedicated page object for this screen at this time.
+			await page.click( 'button[data-e2e-button="decline"]' );
+		} );
+
+		it( 'Confirm plan is purchased', async function () {
+			// No dedicated page object for this screen at this time.
+			await page.waitForSelector( 'h2:has-text("WordPress.com Personal plan")' );
+		} );
+	} );
+
+	describe( 'Launch site', function () {
+		it( 'Navigate to Home dashboard', async function () {
+			const navbarComponent = new NavbarComponent( page );
+			await navbarComponent.clickMySites();
+		} );
+
+		it( 'Verify site is not yet launched', async function () {
+			const testContext = await BrowserManager.browser.newContext();
+			const testPage = await testContext.newPage();
+			// TODO: make a utility to obtain the blog URL.
+			await testPage.goto( `https://${ blogName }.wordpress.com` );
```

Comment:
In this step, instead of clearing the authenticated state for this Page instance I have chosen to launch an entirely new context (private session window). This allows the test spec to check whether the site is hidden behind a `Coming Soon` banner without having to re-log in.

Reply:
In the future, it may be a good idea to have an accessible method within BrowserManager to launch a new context. There exists `launchContext` but I recall that it I could not get it to work in this spec.

Reply:
Do you want to make a task to come back around to do this? I think that would be a worthwhile modularization to make in the future! Looking at the `BrowserManager`, it looks like we _could_ use `launchPage`, but we'd have to do some refactoring around the `Logger`.

Reply:
Sure, let's do that. The `Logger` parameter is really throwing me off, and there were other instances where I could've used the `launchXYZ` methods but was thwarted.

Reply:
Created #56363 for this.